File: frontend/src/api/explain.ts
import { fetchWithAuth } from "./client";

export type FeatureImpact = {
  feature: string;
  value: string;
  impact: number;
  direction: string;
};

export type ExplainResponse = {
  customer_id: string;
  base_value: number;
  final_probability: number;
  features: FeatureImpact[];
};

export async function explainCustomerPrediction(customerId: string): Promise<ExplainResponse> {
  const res = await fetchWithAuth(`/explain/${customerId}`);
  if (!res.ok) {
    const error = await res.json();
    throw new Error(error.detail || "Failed to fetch explanation");
  }
  return res.json();
}

File: backend/app/schemas/explain.py
from pydantic import BaseModel
from typing import List

class FeatureImpact(BaseModel):
    feature: str
    value: str
    impact: float
    direction: str # "positive" (increases churn) or "negative" (decreases churn)

class ExplainResponse(BaseModel):
    customer_id: str
    base_value: float
    final_probability: float
    features: List[FeatureImpact]
